### DISCUSSION TITLE
Source Code not accurate?

### DISCUSSION THREAD

**Discussion by polygonfuture** [srozum/film_camera_tester#8] [Q&A] [answered]
Hi Serhiy,

Can you double check the source code in the github repo?  It appears that both in the latest release and in the main branch of the github there is only the demo test (main.cpp) uploaded.   Is this the full source code for the firmware? 

Unless i'm mistaken this doesnt seem to have any EE testing in it or flash sync tests.  If it is not the latest, do you think you will upload the full firmware at some point to the repo?  

It would be useful in-case some of the components become deprecated and no longer available (such as the TSL237 which is no longer being produced) or users want to utilize a different display like an OLED.

**Comment by srozum** [comment, marked as answer]
Hello,

It's an open hardware project with a free firmware. Source code provided as an example or a template if you'd like to write your own tester with different interface or routines. I want you to understand that main value is an optic-mechanical design of a probes and  light unit.

Hardware used in a project was chosen with careful considerations. All components are available for many years now and still will be obtainable many years later.

**Reply by polygonfuture** [reply]
Ah, I see.  So the firmware is not open source.  Understood.

Just be aware that some of the aliexpress I2C LCD screens come shipped with different I2C adddresses which prevents them from being used with the firmware.
